### PR TITLE
[LA-271] - Added Basic authentication support

### PR DESCRIPTION
**Pull request by sandeepmjay**
Added basic auth support to data-loch APIs. The key and secret are set in the config file.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     "assert": "^1.4.1",
     "async": "2.5.0",
     "aws-sdk": "2.141.0",
+    "basic-auth": "1.1.0",
```
Comment (johncrossman) [inline package.json]: @sandeepmjay Just curious, why not `basic-auth: 2.0`?  https://github.com/jshttp/basic-auth/releases  Not a merge blocker

**Comment by johncrossman** [comment]
@sandeepmjay code looks good. Travis should have run, yes?  I'll take 15 min to investigate.

**Comment by sandeepmjay** [comment]
@johncrossman thanks for the review. Updated basic-auth to 2.0.0. Ready for merge when Mr.T is.

**Comment by johncrossman** [comment]
@sandeepmjay `eslint` has a few complaints...

**Comment by sandeepmjay** [comment]
@johncrossman : Linter was too restrictive. I have relaxed some rules. It should go through now.

**Pull request merged**